storage/blob.go

```diff
@@ -122,6 +125,16 @@ type BlobProperties struct {
 	LeaseStatus           string   `xml:"LeaseStatus"`
 }
 
+// ContentSetting contains various properties of a blob and is an entry
+// in SetBlobProperties
+type ContentSetting struct {
+	ContentMD5      string `xml:"x-ms-blob-content-md5"`
+	ContentLanguage string `xml:"x-ms-blob-content-language"`
+	ContentEncoding string `xml:"x-ms-blob-content-encoding"`
+	ContentType     string `xml:"x-ms-blob-content-type"`
+	CacheControl    string `xml:"x-ms-blob-cache-control"`
```

Comment:
I think the field tag is never used to actually serialize into xml, so we should probably use a tag called "header" for this.